registry/storage/driver/s3-aws/s3.go

```diff
@@ -341,6 +342,23 @@ func FromParameters(parameters map[string]interface{}) (*Driver, error) {
 
 	sessionToken := ""
 
+	virtualHostedStyleBool := false
+	virtualHostedStyle := parameters["virtualHostedStyle"]
+	switch virtualHostedStyle := virtualHostedStyle.(type) {
+	case string:
+		b, err := strconv.ParseBool(virtualHostedStyle)
+		if err != nil {
+			return nil, fmt.Errorf("the virtualHostedStyle parameter should be a boolean")
+		}
+		virtualHostedStyleBool = b
+	case bool:
+		virtualHostedStyleBool = virtualHostedStyle
+	case nil:
+		// do nothing
+	default:
+		return nil, fmt.Errorf("the virtualHostedStyle parameter should be a boolean")
+	}
+
```

Comment:
nit - move the entire block above `sessionToken := ""`

Reply:
updated in next push